### PR TITLE
ursadb: init at v1.2.0

### PR DESCRIPTION
**Pull request by msm-code**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux) (AFAIU it's the default on nixos)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` (n/a - no packages are dependent on this yet. Tried it, "No diff detected, stopping review..."?)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Emantor** [comment]
Throws an error when I try to run it:
```
nix-shell:~/.cache/nixpkgs-review/pr-83905/results/ursadb/bin]$ ./ursadb_new ~/tmp/mydb

[nix-shell:~/.cache/nixpkgs-review/pr-83905/results/ursadb/bin]$ ./ursadb ~/tmp/mydb 2048
terminate called after throwing an instance of 'zmq::error_t'
  what():  Invalid argument
Aborted (core dumped)
```

**Comment by msm-code** [comment]
Hi,

thanks for input, If I knew anyone will test it I'd add more detailed instructions .

Looks like there's one parameter too much:

```
msm@scourge ~/P/nixpkgs> nix-build -A ursadb
/nix/store/8xlnxlm8q8y5bilzwkxviaxzi8cgya1h-ursadb
msm@scourge ~/P/nixpkgs> ./result/bin/ursadb_new /tmp/ursa/test.db
msm@scourge ~/P/nixpkgs> ./result/bin/ursadb /tmp/ursa/test.db
```

And there should be a process listening on `:9281`

I use https://github.com/CERT-Polska/ursadb-cli and https://github.com/CERT-Polska/mquery for testing (and on production), though they're not on nixpkgs (yet?).

**Comment by Emantor** [comment]
> Hi,
> 
> thanks for input, If I knew anyone will test it I'd add more detailed instructions .
> 
> Looks like you've addded one parameter too much (if you want to specify the bind address you can do it like `tcp://127.0.0.1:9281` - the error message could've been clearer i guess).
> 
> ```
> msm@scourge ~/P/nixpkgs> nix-build -A ursadb
> /nix/store/8xlnxlm8q8y5bilzwkxviaxzi8cgya1h-ursadb
> msm@scourge ~/P/nixpkgs> ./result/bin/ursadb_new /tmp/ursa/test.db
> msm@scourge ~/P/nixpkgs> ./result/bin/ursadb /tmp/ursa/test.db
> ```
> 
> And there should be a process listening on `:9281`

Indeed, somehow I got `[bind_address]` confused with `[socket]` and supplied only a number. With your instructions it starts correctly. 

> I use https://github.com/CERT-Polska/ursadb-cli and https://github.com/CERT-Polska/mquery for testing (and on production), though they're not on nixpkgs (yet?).

Future packaging opportunity? :grin:

**Comment by msm-code** [comment]
`Future packaging opportunity?`

I'm very much happy to to this in the future (though this service is the `core` so prerequisite for the others).

Maybe not ursadb-cli (I've noticed python package maintainers don't like too niche packages due to maintenance burden), but certainly mquery (right now the only supported installation method is docker, which is very meh).

File: pkgs/servers/ursadb/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  buildInputs = [
+    cmake
+    zeromq
+    cppzmq
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/CERT-Polska/ursadb;
```
Comment (Emantor) [inline pkgs/servers/ursadb/default.nix:26]:
```suggestion
    homepage = "https://github.com/CERT-Polska/ursadb";
```

File: pkgs/servers/ursadb/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl, cmake, zeromq, cppzmq }:
+
+stdenv.mkDerivation {
+  name = "ursadb";
+  version = "1.2.0";
+
+  src = fetchurl {
+    url = https://github.com/CERT-Polska/ursadb/archive/v1.2.0.tar.gz;
```
Comment (Emantor) [inline pkgs/servers/ursadb/default.nix:8]:
```suggestion
    url = "https://github.com/CERT-Polska/ursadb/archive/v1.2.0.tar.gz";
```

**Review by Emantor — CHANGES_REQUESTED** [review]

**Review by Emantor — APPROVED** [review]

**Comment by Emantor** [comment]
Would you kindly squash down the review commits into the `init at` commit? Than this should be ready to go.

**Comment by msm-code** [comment]
Thanks for your suggestions! The url thing was accidental, I didn't even know that I can have strings without quotes (I assumed that quotes make a string, no quotes but slash == path, variable otherwise).

**Comment by msm-code** [comment]
Sure, I was not sure if I was supposed to squash it or leave as it is

**Comment by Emantor** [comment]
> Sure, I was not sure if I was supposed to squash it or leave as it is

The [submission guidelines](https://nixos.org/nixpkgs/manual/#chap-submitting-changes) mention that review fixes should be squashed.

**Comment by msm-code** [comment]
You're right, I missed that. FWIW I squashed my changes before the first push. Thanks for clarifying!

File: pkgs/servers/ursadb/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchurl {
+    url = "https://github.com/CERT-Polska/ursadb/archive/v1.2.0.tar.gz";
+    sha256 = "10dax3mswq0x4cfrpi31b7ii7bxl536wz1j11b7f5c0zw9pjxzym";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ursadb $out/bin/
+    cp ursadb_new $out/bin/
+    cp ursadb_trim $out/bin/
```
Comment (bhipple) [inline pkgs/servers/ursadb/default.nix:16]: You could do this on one like, e.g.

```
cp ursadb ursadb_new ursadb_trim $out/bin/
```

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 83905` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- ursadb
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**